### ISSUE TITLE
Support document middleware

### ISSUE THREAD

**Issue by AdamGerthel** [feathersjs-ecosystem/feathers-mongoose#395]
## Feature request

Support document middleware

## Background

AFAIK, document middleware, such as [mongoose pre-save hooks](https://mongoosejs.com/docs/middleware.html#pre), are not supported. I'm assuming that this is because feathers-mongoose uses queries such as `findOneAndUpdate()` rather than loading the document and then saving it (see https://mongoosejs.com/docs/middleware.html#notes). 

As suggested in other issues, it's possible, and [often recommended](https://github.com/feathersjs-ecosystem/feathers-mongoose/issues/226#issuecomment-354664906), to utilise [Feathers service hooks](https://docs.feathersjs.com/api/hooks.html) instead. However, there are use cases when that option is not very fitting. An example is when you are re-using a single schema in several other schemas. Perhaps even nested on different levels in different models.

In such a use case, using service hooks would be very messy compared to a single, all-encompassing Mongoose pre validate/save/remove hook.

**Comment by daffl** [comment]
This should work already as long as you set the [`lean` option](https://github.com/feathersjs-ecosystem/feathers-mongoose#serviceoptions) to `false`.

**Comment by AdamGerthel** [comment]
@daffl I've read that too somewhere, but it doesn't! I've tried every possible way/setting/option.

**Comment by daffl** [comment]
Ideally we'd need an example of what isn't working as expected then.

**Comment by AdamGerthel** [comment]
Yep, I totally get that. Will see if I get time to throw together a small repro repo.
